lib/core/nav/HeaderNav.js

```diff
@@ -126,31 +134,24 @@ class HeaderNav extends React.Component {
         : '';
       const versionPart =
         env.versioning.enabled && this.props.version !== 'next'
-          ? 'version-' +
+          ? (this.props.language || 'en') +
+            '-version-' +
             (this.props.version || env.versioning.latestVersion) +
             '-'
           : '';
       const id = langPart + versionPart + link.doc;
       if (!Metadata[id]) {
-        let errorStr =
-          "Processing the following `doc` field in `headerLinks` within `siteConfig.js`: '" +
-          link.doc +
-          "'";
-        if (id === link.doc) {
-          errorStr +=
-            ' It looks like there is no document with that id that exists in your docs directory. Please double check the spelling of your `doc` field and the `id` fields of your docs.';
-        } else {
-          errorStr +=
-            '. Check the spelling of your `doc` field. If that seems sane, and a document in your docs folder exists with that `id` value, \nthen this is likely a bug in Docusaurus.' +
-            ' Docusaurus thinks one or both of translations (currently set to: ' +
-            env.translation.enabled +
-            ') or versioning (currently set to: ' +
-            env.versioning.enabled +
-            ") is enabled when maybe they should not be. \nThus my internal id for this doc is: '" +
-            id +
-            "'. Please file an issue for this possible bug on GitHub.";
+        if (id != link.doc) {
+          throw new Error(
+            "It looks like you've enabled language support, but haven't provided translated files. The document with id: '" +
+              id +
+              "' doesn't exist."
+          );
         }
```

Comment:
This must be a remnant from another pull request. Is it a bad change?

Reply:
I added this in a previous commit.